### PR TITLE
Virtualbox: allow use of vagrant default synced folders implementation

### PR DESCRIPTION
**Pull request by devste**
This feature allows to use the default synced folder implementation. Up to now, only nfs was used, which could cause a problem, depending on the environment. The implementation can be set in parameters.yml. To stay backward compatible, the default option (if not configured) is still nfs.

* This PR is a : [New feature]

- [ ] Documentation is written - no documentation about the current nfs requirements anywhere.
- [x] `parameters.yml.dist` is updated
- [ ] `playbook.yml.dist` is updated - not applicable
- [x] [Changelog](https://github.com/liip/drifter/blob/master/CHANGELOG.md) was updated

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -78,14 +78,18 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     config.vm.provider "virtualbox" do |v, override|
         override.vm.network :private_network, ip: custom_config.get('box_ip', "10.10.10.10")
 
-        if Vagrant::Util::Platform.windows? && !Vagrant.has_plugin?("vagrant-winnfsd")
-            puts "*************************************************************************"
-            puts "Please install the plugin vagrant-winnfsd to have NFS Support on Windows!"
-            puts "vagrant plugin install vagrant-winnfsd"
-            puts "*************************************************************************"
-            exit 1
+        sync_type = custom_config.get('sync_type')
```
Comment (sephii) [inline Vagrantfile]: I would name name this `synced_folder_type`.

File: parameters.yml.dist [hunk 2]
```diff
@@ -28,6 +28,10 @@ database_name: "example"
 # Root directory of your project for the webserver and other purposes
 root_directory: "/vagrant/"
 
+# Synced folder type to use. Only "nfs" and "" will work. Default in drifter
```
Comment (sephii) [inline parameters.yml.dist]: I would use values `nfs` and `virtualbox` to be more explicit. Also the comment mentions that only nfs and "" will work, but actually any value will work. I think we should validate this in the `Vagrantfile` (ie. make sure value is either `nfs` or `virtualbox`). Could you add that?

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ For migration information, you can always have a look at https://liip-drifter.re
 ## Unreleased
 
 ### Added
+- Virtualbox: allow use of the vagrant default synced folders implementation.
```
Comment (sephii) [inline CHANGELOG.md]: Could you please mention "VirtualBox shared folders" instead of "default synced folders"?

**Comment by devste** [comment]
I completely agree with the suggestions, so I just updated the files.

File: Vagrantfile [hunk 4]
```diff
@@ -46,6 +46,15 @@ else
...
 
+synced_folder_type = custom_config.get('synced_folder_type')
+
+unless (synced_folder_type == "nfs" || synced_folder_type == "virtualbox")
+    class SyncedFolderTypeError < Vagrant::Errors::VagrantError
+        self.error_message "[synced_folder_type] Only 'nfs' and 'virtualbox' are supported values"
+    end
+    raise SyncedFolderTypeError
+end
+
```
Comment (devste) [inline Vagrantfile]: The suggested way by Vagrant to raise expected errors: https://github.com/mitchellh/vagrant/blob/master/lib/vagrant/errors.rb
I could not subclass Vagrant::Errors::ConfigInvalid as this causes an i18n error on my machine (vagrant up does depend on the host) and I don't have time to investigate.

**Comment by sephii** [comment]
Thanks! 🐢

**Pull request merged**